bolt-sidecar/src/config/commitments.rs

```diff
@@ -0,0 +1,64 @@
+use clap::{ArgGroup, Args};
+use reqwest::Url;
+use serde::Deserialize;
+
+use crate::common::secrets::EcdsaSecretKeyWrapper;
+
+/// Default port for the JSON-RPC server exposed by the sidecar supporting the Commitments API.
+///
+/// 8017 -> BOLT :)
+pub const DEFAULT_RPC_PORT: u16 = 8017;
+
+/// Command-line options for signing constraint messages
+#[derive(Args, Deserialize, Debug, Clone)]
+#[clap(
+    group = ArgGroup::new("commitments-opts").required(true)
+        .args(&["commitments_port", "firewall_rpc"])
+)]
+pub struct CommitmentOpts {
+    /// Port to listen on for incoming JSON-RPC requests of the Commitments API.
+    /// This port should be open on your firewall in order to receive external requests!
+    #[clap(long, env = "BOLT_SIDECAR_PORT", default_value = stringify!(DEFAULT_RPC_PORT))]
```

Comment:
Yes, this means if you spin up the sidecar without specifying any "gateway" or whatever it will be started in "server" mode.

Reply:
No, it means if you spin up the sidecar specifying a list of gateways it will be started in server mode anyway because there is a default value. that's what I understood from https://github.com/chainbound/bolt/pull/552/files#diff-d9dd450d19da416698fd8f0b096f157f4a2c36f99146757f589dde830b36adceR231

Reply:
Right, good catch. Then I'll drop any default behaviour so that one of the two is mandatory and that's it.

Reply:
this is why I proposed the USE_GATEWAYS variable, it's similar to how in geth you have to use 
`--rpc --rpc.addr 0.0.0.0 --rpc-port 8545` as in:

* `--rpc` enables the feature
* `--rpc.addr` and `.port` configure the rpc feature

in our case:
USE_GATEWAYS=true will dispatch the websocket provider etc
USE_GATEWAYS=false will always use the configured server port: default 8017 and can be configured

Reply:
I get the idea and I think it's works great if you're enabling features that are independent between each other.
In this context the "USE" flag will also have the additional semantics of invalidating another flag, which is fine but I'll try to avoid as possible.

Since when starting up the sidecar we have to distinguish between the two running modes with an `if` condition, what do you think to add a default there for the port? Like so:
```rs
        let api_events_rx = if let Some(urls) = opts.commitment_opts.gateway_urls.clone() {
            CommitmentsFirewallRecv::new(
                opts.commitment_opts.operator_private_key.clone(),
                opts.chain.chain,
                urls,
            )
            .run()
            .await
        } else {
            let port = opts.commitment_opts.port.unwrap_or(DEFAULT_RPC_PORT);
            // start the commitments api server
            let api_addr = format!("0.0.0.0:{}", port);
            let (api_events_tx, api_events_rx) = mpsc::channel(1024);
            CommitmentsApiServer::new(api_addr).run(api_events_tx, opts.limits).await;
            api_events_rx
        };
```

This would result in:
- you can either choose between `PORT` and `GATEWAY_URLS`
- if none of them are set, the default is `PORT=8017`.

Reply:
I like this a lot more, thanks!

Reply:
Great, happy we've find a good design after some iterations :)